bindings/pydeck-carto/pydeck_carto/carto_auth/layer_validator.py

```diff
@@ -0,0 +1,64 @@
+from pydeck import Layer, types
+from pydeck_carto.carto_auth.auth import CartoAuth
+from pydeck_carto.layer import MapType, CartoConnection, GeoColumnType
+
+
+def validate_attr(layer, attr, param_type=types.String):
+    if not hasattr(layer, attr):
+        raise ValueError(f'Missing parameter {attr} on the layer')
+    attr_value = getattr(layer, attr)
+    if not attr_value:
+        raise ValueError(f'Parameter {attr} not valid')
+    if not isinstance(attr_value, param_type):
+        raise ValueError(f'Parameter {attr} must be a {param_type}')
+
+
+def is_valid_carto_layer(layer: Layer, carto_auth: CartoAuth):
+    if layer.type != 'CartoLayer':
+        raise ValueError('The layer type must be "CartoLayer"')
+
+    validate_attr(layer, 'credentials', dict)
+    validate_attr(layer, 'data', str)
+
+    required_strings = ['type_', 'connection']
```

Comment:
Why the loop instead of

```
validate_attr(layer, 'type_')
validate_attr(layer, 'connection')
```

Reply:
To group all the fields that require `types.String` type as parameters, but maybe as it is only 2 it makes sense to follow the same pattern as the previous validations, I'll change it